suite/tests/api/opnd-a64.c

```diff
@@ -303,21 +307,479 @@ test_opnd_invert_immed_int()
+
+    default: ASSERT(false);
+    }
+}
+
+/* Used by test_compute_vector_address() to determine whether an instruction reads or
+ * writes its memory operand and set test expectations.
+ * This isn't an exhaustive list of opcodes; it just contains the ones used in the test
+ */
+static opnd_size_t
+op_mem_size(int op)
+{
+    switch (op) {
+    case OP_ld1b:
+    case OP_ldnt1b:
+    case OP_st1b:
+    case OP_stnt1b: return OPSZ_1;
+    case OP_ld1h:
+    case OP_ldnt1h:
+    case OP_st1h:
+    case OP_stnt1h: return OPSZ_2;
+    case OP_ld1w:
+    case OP_ldnt1w:
+    case OP_st1w:
+    case OP_stnt1w: return OPSZ_4;
+    case OP_ld1d:
+    case OP_ldnt1d:
+    case OP_st1d:
+    case OP_stnt1d: return OPSZ_8;
+
+    default: ASSERT(false);
+    }
+}
+
+void
+test_compute_vector_address(void *drcontext)
+{
+    const int original_vector_length = dr_get_sve_vector_length();
+    ASSERT(dr_set_sve_vector_length(256));
+
+#define SCALAR_BASE_REG 0
+
+#define INDEX_REG_D 0
+#define INDEX_REG_S 1
+#define BASE_REG_D 2
+#define BASE_REG_S 3
+
+    dr_mcontext_t mc = {
+        .size = sizeof(dr_mcontext_t),
+        .flags = DR_MC_ALL,
+        .r0 = 0x8000000000000000, /* SCALAR_BASE_REG */
+        .r1 = 1,
+        .r2 = 2,
+        .r3 = 3,
+        .r4 = 4,
+        .r5 = 5,
+        .r6 = 6,
+        .r7 = 7,
+        .r8 = 0xffffffffffffffff,
+        .simd[INDEX_REG_D].u64 = { 0x0000000000010000, 0x0000000000020000,
+                                   0xffffffffffff0000, 0xfffffffffffe0000 },
+        .simd[INDEX_REG_S].u32 = { 0x00010000, 0x00020000, 0x00030000, 0x00040000,
+                                   0xffff0000, 0xfffd0000, 0xfffc0000, 0xfffb0000 },
+        .simd[BASE_REG_D].u64 = { 0x0000000000000000, 0x8000000000000000,
+                                  0xffffffffffffffff, 0x0000000010000000 },
+        .simd[BASE_REG_S].u32 = { 0x00000000, 0x80000000, 0xffffffff, 0x00010000,
+                                  0x10000000, 0x20000000, 0x30000000, 0x40000000 },
+    };
+
+    for (size_t i = BASE_REG_S + 1; i < MCXT_NUM_SIMD_SVE_SLOTS; i++) {
+        static const uint64 poison[4] = { 0xdeaddeaddeaddead, 0xdeaddeaddeaddead,
+                                          0xdeaddeaddeaddead, 0xdeaddeaddeaddead };
+        memcpy(&mc.simd[i].u64[0], poison, sizeof(poison));
+    }
+    for (size_t i = 0; i < MCXT_NUM_SVEP_SLOTS; i++) {
+        mc.svep[i].u32[0] = 0xffffffff;
+    }
+
+/* Map SVE element sizes to opnd_size_t */
+#define ELSZ_B OPSZ_1
+#define ELSZ_H OPSZ_2
+#define ELSZ_S OPSZ_4
+#define ELSZ_D OPSZ_8
+
+#define EXPECT(...)                                \
+    app_pc addresses[] = { __VA_ARGS__ };          \
+    vector_address_test_expectation_t expected = { \
+        addresses,                                 \
+        sizeof(addresses) / sizeof(app_pc),        \
+        false,                                     \
+    }
+
+#define VEC_ADDR_TEST(op, pg, mask, create_mem_opnd, decl_expect)                       \
```

Comment:
What is "pg"?  Is there a more descriptive name to use?

Reply:
Its the instruction `Pg` operand (governing predicate register). I have changed it to "governing_pred_reg".